interfaces/kratos_external_idp/v0/schema.py

```diff
@@ -0,0 +1,125 @@
+"""This file defines the schemas for the provider and requirer sides of the kratos_external_idp interface.
+
+It exposes two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+
+Examples:
+    ProviderSchema:
+        unit: <empty>
+        app: {
+          "providers":[
+            "client_id": "client_id"
+            "client_secret": "cl1ent-s3cRet"
+            "secret_backend": "relation"
+            "tenant_id": "4242424242424242"
+            "provider": "microsoft"
+          ]
+        }
+
+    RequirerSchema:
+        unit: <empty>
+        app: {
+          "providers": [
+            "redirect_uri": "https://example.kratos.com/self-service/methods/oidc/callback/microsoft"
+            "provider_id": "microsoft"
+          ]
+        }
+"""
+
+from enum import Enum
+from typing import List, Optional
+from pydantic import AnyHttpUrl, BaseModel, validator
+
+from interface_tester.schema_base import DataBagSchema
+
+
+class Url(BaseModel):
+    url: AnyHttpUrl
+
+
+class Provider(Enum):
+    microsoft = "microsoft"
+    apple = "apple"
+    generic = "generic"
+    auth0 = "auth0"
+    google = "google"
+    facebook = "facebook"
+    github = "github"
+    gitlab = "gitlab"
+    slack = "slack"
+    spotify = "spotify"
+    discord = "discord"
+    twitch = "twitch"
+    netid = "netid"
+    yandex = "yandex"
+    vk = "vk"
+    dingtalk = "dingtalk"
+
+
+class SecretBackend(Enum):
+    relation = "relation"
+    secret = "secret"
+    vault = "vault"
+
+
+class ExternalIdpProvider(BaseModel):
+    client_id: str
+    client_secret: str
+    secret_backend: SecretBackend = "relation"
+    provider: Provider
+    scope: Optional[str]
+    provider_id: Optional[str]
+    jsonnet_mapper: Optional[str]
+    tenant_id: Optional[str]
+    private_key: Optional[str]
+    private_key_id: Optional[str]
+    team_id: Optional[str]
+    issuer_url: Optional[str]
+
+    @validator("tenant_id")
+    def provider_must_be_microsoft(cls, v, values):
```

Comment:
Why do we only have those validators for microsoft and apple but not for others? Similar to Jon's comment above, if those are the only ones we support and test, let's not include the ones we don't.

Reply:
Those validators are there because these are "special" providers, ie they need some extra configuration to work.